altapay.php

```diff
@@ -0,0 +1,2353 @@
+require_once dirname(__FILE__) . '/classes/MerchantAPI.php';
+require_once dirname(__FILE__) . '/helpers.php';
+
+class ALTAPAY extends PaymentModule
+{
+    public $url;
+    public $captureStatus;
+    public $terminal;
+    public $username;
+    public $password;
+    public $payment_type;
+    private $Mhtml = '';
+    private $postErrors = array();
+    private $paymentMethodIconDir = 'views/img/payment_icons';
+
+    public function __construct()
+    {
+        $this->name = 'altapay';
+        $this->tab = 'payments_gateways';
+        $this->version = '3.1.0';
+        $this->v16 = _PS_VERSION_ >= '1.6.1.24';
+        $this->v17 = _PS_VERSION_ >= '1.7.6.5';
+        $this->author = 'Altapay A/S';
+        $this->is_eu_compatible = 1;
+        $this->ps_versions_compliancy = array('min' => '1.6.1.24', 'max' => '1.7.6.5');
+
+        $this->currencies = true;
+        $this->currencies_mode = 'checkbox';
+
+        $this->bootstrap = true;
+
+        $config = Configuration::getMultiple(array(
+            'ALTAPAY_USERNAME',
+            'ALTAPAY_PASSWORD',
+            'ALTAPAY_URL',
+            'AUTOCAPTURE_STATUSES',
+            'ALTAPAY_TERMINAL'
+        ));
+        if (isset($config['ALTAPAY_USERNAME'])) {
+            $this->username = $config['ALTAPAY_USERNAME'];
+        }
+        if (isset($config['ALTAPAY_PASSWORD'])) {
+            $this->password = $config['ALTAPAY_PASSWORD'];
+        }
+        if (isset($config['ALTAPAY_URL'])) {
+            $this->url = $config['ALTAPAY_URL'];
+        }
+        if (isset($config['AUTOCAPTURE_STATUSES'])) {
+            $this->captureStatus = $config['AUTOCAPTURE_STATUSES'];
+        }
+
+
+        parent::__construct();
+
+        $this->displayName = $this->l('Altapay for Prestashop');
+        $this->description = $this->l('Altapay for Prestashop - Payments less complicated');
+        $this->confirmUninstall = $this->l('Are you sure about removing these details?');
+
+        // Make sure currencies are configured for this payment module
+        if (!count(Currency::checkPaymentCurrencies($this->id))) {
+            $this->warning = $this->l('No currency has been set for this module.');
+        }
+    }
+
+    /**
+     * Called on install
+     */
+    public function install()
+    {
+        if (!parent::install()
+            || !$this->registerHook('payment')
+            || !$this->registerHook('paymentOptions')
+            || !$this->registerHook('paymentReturn')
+            || !$this->registerHook('adminOrder')
+            || !$this->registerHook('actionOrderStatusUpdate')
+            || !$this->registerHook('displayBackOfficeHeader')
+            || !$this->registerHook('displayCustomerAccount')
+            || !$this->registerHook('actionFrontControllerSetMedia')
+            || !$this->registerHook('actionOrderStatusPostUpdate')
+        ) {
+            return false;
+        }
+
+        /* This table captures the payment information */
```

Comment:
In most other parts of the code online comments appear to be done using //
```suggestion
        // This table captures the payment information
```